app/internal/dynamo/client.go

```diff
@@ -63,7 +69,7 @@ func (c *Client) Get(ctx context.Context, pk, sk string, v interface{}) error {
 		return err
 	}
 	if result.Item == nil {
-		return nil
+		return &NotFoundError{}
```

Comment:
since you defined `Error` on the non-pointer receiver you could have returned `NotFoundError{}` here